src/t2iapi/device/service.proto

```diff
@@ -57,13 +57,21 @@ service DeviceService {
     rpc SetLanguage (t2iapi.device.SetLanguageRequest) returns (BasicResponse);
 
     /*
-    Get all descriptor handles which can be removed and reinserted into the MDIB of the device.
+    Get all descriptor handles which can be removed and reinserted into the MDIB of the device
+    and whose descriptors are of the given class.
     Includes handles which have been removed already and can be reinserted.
     Handles must stay the same once reinserted into the MDIB.
-    The handles shall be representative of the devices capabilities to remove
-    descriptors (at least one of every possible kind).
+
+    Filtering by class should be equivalent to doing it after the manipulation:
+    i.e. the handles resulting from calling the GetRemovableDescriptorsOfClass manipulation with
+    descriptor_class = X should be the same as those resulting from calling the GetRemovableDescriptorsOfClass
+    manipulation without filtering and then applying .stream().filter((Object x) -> x instanceof X') to
```

Comment:
No java examples in the generic api documentation, this does not help Python developers

Reply:
Revised.